lib/cylc/prerequisite.py

```diff
@@ -223,6 +224,45 @@ def dump(self):
         # (Else trigger wiped out by pre-initial simplification.)
         return res
 
+    def api_dump(self):
+        """Return list of populated Protobuf data objects."""
+        if self.conditional_expression:
+            temp = self.get_raw_conditional_expression()
+            temp = temp.replace('|', ' | ')
+            temp = temp.replace('&', ' & ')
+        else:
+            for s_msg in self.satisfied:
+                temp = self.MESSAGE_TEMPLATE % s_msg
+        conds = []
+        num_length = int(math.ceil(float(len(self.satisfied)) / float(10)))
+        for ind, message_tuple in enumerate(sorted(self.satisfied)):
+            t_id = '%s.%s' % message_tuple[0:2]
+            char = 'c%.{0}d'.format(num_length) % ind
+            c_msg = self.MESSAGE_TEMPLATE % message_tuple
+            c_val = self.satisfied[message_tuple]
+            c_bool = bool(c_val)
+            if c_val is False:
+                c_val = "unsatisfied"
+            cond = PbCondition(
+                task_id=t_id,
+                task_proxy=t_id,
+                expr_alias=char,
+                req_state=message_tuple[2],
+                satisfied=c_bool,
+                message=c_val,
+            )
+            conds.append(cond)
+            temp = temp.replace(c_msg, char)
```

Comment:
Will test it later if possible, but if `temp` was not created before in the first `if/else`, it would fail when the code got here.

Reply:
Don't think it's possible as `self.satisfied` is looped over (in different ways) in both places. For the else case it will be just a single item.

Thanks this reminded me to update the `t_id` and I got rid of the duplicate (it was in case just the id was desired)..